js_modules/dagster-ui/packages/ui-core/src/pipelines/useOpGraphSelectionAutoCompleteProvider.tsx

```diff
@@ -0,0 +1,49 @@
+import {useMemo} from 'react';
+
+import {GraphQueryItem} from '../app/GraphQueryImpl';
+import {iconMap} from '../gantt/useGanttChartSelectionAutoCompleteProvider';
+import {createSelectionAutoComplete} from '../selection/SelectionAutoComplete';
+import {
+  SelectionAutoCompleteProvider,
+  createProvider,
+} from '../selection/SelectionAutoCompleteProvider';
+
+export const useOpGraphSelectionAutoCompleteProvider = (
+  items: GraphQueryItem[],
+): SelectionAutoCompleteProvider => {
+  const attributesMap = useMemo(() => {
+    const names = new Set<string>();
+    items.forEach((item) => {
+      names.add(item.name);
+    });
+    return {name: Array.from(names)};
+  }, [items]);
+
+  const baseProvider = useMemo(
+    () =>
+      createProvider({
+        attributesMap,
+        primaryAttributeKey: 'name',
+        attributeToIcon: iconMap,
+      }),
+    [attributesMap],
+  );
+  const selectionHint = useMemo(() => createSelectionAutoComplete(baseProvider), [baseProvider]);
+
+  return useMemo(
+    () => ({
+      ...baseProvider,
+      useAutoComplete: ({line, cursorIndex}) => {
```

Comment:
It might be for cloud, but this function is identical in all three of the hook implementations - I wonder if the hook could provide selectionHint and the remainder could be in the callsite? Will read on :-)

Reply:
hmm, yeah I didn't think too much about it if something strikes me I'll do it but for now its a very small amount of duplication